DUI3-DX/Converters/Autocad/Speckle.Converters.AutocadShared/ToSpeckle/Geometry/Polyline2dToSpeckleConverter.cs

```diff
@@ -0,0 +1,234 @@
+        break;
+      case ADB.Poly2dType.QuadSplinePoly:
+        polyType = SOG.Autocad.AutocadPolyType.QuadSpline2d;
+        break;
+    }
+
+    // get all vertex data
+    List<double> value = new();
+    List<double> bulges = new();
+    List<double> tangents = new();
+    List<ADB.Vertex2d> vertices = target
+      .GetSubEntities<ADB.Vertex2d>(
+        ADB.OpenMode.ForRead,
+        _contextStack.Current.Document.TransactionManager.TopTransaction
+      )
+      .ToList();
+
+    for (int i = 0; i < vertices.Count; i++)
+    {
+      ADB.Vertex2d vertex = vertices[i];
+
+      // get vertex value in the Global Coordinate System (GCS).
+      value.AddRange(vertex.Position.ToArray());
+
+      // get the bulge and tangent, and 3d point for displayvalue
+      bulges.Add(vertex.Bulge);
+      tangents.Add(vertex.Tangent);
+
+      // POC: check this data is necessary on receive!
+      // construct the spline curve segment
+      // switch (vertex.VertexType)
+      // {
+      //   case Vertex2dType.CurveFitVertex:
+      //     break;
+      //   case Vertex2dType.SplineFitVertex:
+      //     break;
+      //   case Vertex2dType.SplineControlVertex:
+      //     break;
+      // }
+    }
+
+    List<Objects.ICurve> segments = new();
+
+    // POC: retrieve spline display value here for database-resident polylines by connecting all vertex points
+    if (target.Database is not null)
+    {
+      var exploded = new ADB.DBObjectCollection();
+      target.Explode(exploded);
+      AG.Point3d previousPoint = new();
+      for (int i = 0; i < exploded.Count; i++)
+      {
+        var segment = exploded[i] as ADB.Curve;
+
+        if (segment == null)
+        {
+          continue;
+        }
+
+        if (i == 0 && exploded.Count > 1)
+        {
+          // get the connection point to the next segment - this is necessary since imported polycurves might have segments in different directions
+          var nextSegment = exploded[i + 1] as ADB.Curve;
+          if (nextSegment == null)
+          {
+            continue;
+          }
+          AG.Point3d connectionPoint =
+            nextSegment.StartPoint.IsEqualTo(segment.StartPoint) || nextSegment.StartPoint.IsEqualTo(segment.EndPoint)
+              ? nextSegment.StartPoint
+              : nextSegment.EndPoint;
+
+          previousPoint = connectionPoint;
+          segment = GetCorrectSegmentDirection(segment, connectionPoint, true, out AG.Point3d _);
+        }
+        else
+        {
+          segment = GetCorrectSegmentDirection(segment, previousPoint, false, out previousPoint);
+        }
+
+        switch (segment)
+        {
+          case ADB.Arc arc:
+            segments.Add(_arcConverter.RawConvert(arc));
+            break;
+          case ADB.Line line:
+            segments.Add(_lineConverter.RawConvert(line));
+            break;
+          case ADB.Spline spl:
+            segments.Add(_splineConverter.RawConvert(spl));
+            break;
+        }
```

Comment:
We could do with a POC comment or something about the skipping of the segment if this doesn't match - should we skip, even if this works?